### DISCUSSION TITLE
IDE settings

### DISCUSSION THREAD

**Discussion by Morbidous** [MNES-Emulator/MNES#16] [Q&A] [answered]
I was going to ask about specific issues, but it's way too much work to go over a single IDE setting in detail.

The main issue is that we should use `.editorconfig` to declare the IDE preferences for the solution. I need to do this to avoid changing my own settings which are synced with my account and make it to work when I change them. But I need to get rid of the IDE warnings that my settings are causing, so the settings need to be set in `.editorconfig` instead. I will start by choosing what I think are sane settings, but feel free (or encouraged) to change things as you prefer, I am just setting them so that I can keep moving forward.

I will mention some that I am changing here, maybe, although that might be too much work. I won't layout the whole meaning of the setting. IDE0007 and IDE0008 are about `var` usage. I'm going to prefer to use `var` everywhere. (That is already what I do but I will put it explicitly in the `.editorconfig`.) I will then update places in the code accordingly. If you want this to be different feel free to change it.

**Comment by Morbidous** [comment]
An assumption here is that our IDEs will actually obey the setting which I haven't confirmed yet. Also I have to learn how `.editorconfig` even works. Sigh.

**Comment by Morbidous** [comment]
Trailing commas is another one. I don't usually use them but I don't really have a preference. I do know they have benefits. I will turn them on (if I can) in `.editorconfig` which is different than my own setting. Hmm, I don't think this one is possible yet. Well there are JetBrains specific settings, I'll have to use that.

**Comment by Morbidous** [comment]
K that gets my warning count to zero, so I guess it wasn't too bad, just a couple settings and some refactoring.

**Comment by jamieyello** [comment, marked as answer]
Yeah, editorconfig will work in the IDE that I use and in most IDE's by default (unless you're a unique person using VIM)

While you're messing with it, one thing I will say is that I find it cluttering when this

```
public Thing(int i, float f, string s) {
    // stuff
}
```

is formatted as this

```
public Thing(
    int i,
    float f,
    string s
)  {
    // stuff
}
```

That's my only feedback so far, I find that breaks the readability/line count pretty hard, especially when there are many parameters. I found I had to undo some constructor classes formatted this way to make it easier to see the whole thing at once.

**Reply by Morbidous** [reply]
Ok I will try to avoid parameters stacking
